src/beamlime/applications/base.py

```diff
@@ -206,25 +149,17 @@ async def route(self, message: MessageProtocol) -> None:
         for result in results:
             self.message_pipe.put(result)
 
-    async def run(self) -> None:
+    async def run(
+        self,
+    ) -> AsyncGenerator[Optional[MessageProtocol], None]:
         """Message router daemon."""
-        data_monitor = self.data_pipe_monitor(
-            self.message_pipe,
-            timeout=0.1,
-            interval=0.1,
-            size_monitor_func=self.message_pipe.qsize,
-        )
-
-        while not self._break_routing_loop:
-            if await data_monitor():
```

Comment:
So you removed the monitor and the check for the number of messages. Why is this not needed any more?

Reply:
Because message router doesn't decide when to route/stop anymore.
It's a generator so the ``Application`` instance will call ``next`` on it.